app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt

```diff
@@ -192,6 +193,21 @@ class GleanMetricsService(private val context: Context) : MetricsService {
     }
 
     override fun shouldTrack(event: Event): Boolean {
+        if (event is Event.PreferenceToggled) {
+            return switchPreferenceTelemetryAllowList.contains(event.preferenceKey)
+        }
+
         return event.wrapper != null
     }
+
+    // Switch preferences that should have their toggle events sent to Glean
+    private val switchPreferenceTelemetryAllowList = listOf(
```

Comment:
I would move this to the PreferenceToggled type under a private failable constructor to keep all of the logic around that type together :+1: